views/js/qtiCreator/widgets/interactions/hotspotInteraction/states/Map.js

```diff
@@ -61,93 +76,98 @@ define([
-        commonRenderer.setResponse(interaction, PciResponse.serialize(_.values(response.getCorrect()), interaction));   
+        commonRenderer.setResponse(interaction, PciResponse.serialize(_.values(response.getCorrect()), interaction));
     };
 
     /**
      * Exit the map state
      */
-    var exitMapState = function exitMapState(){
-        var widget = this.widget;
-        var interaction = widget.element;
-        
-        if(!interaction.paper){
+    const exitMapState = function exitMapState() {
+        const widget = this.widget;
+        const interaction = widget.element;
+
+        if (!interaction.paper) {
             return;
         }
 
         //destroy the common renderer
         commonRenderer.resetResponse(interaction);
-        commonRenderer.destroy(interaction); 
+        commonRenderer.destroy(interaction);
         instructionMgr.removeInstructions(interaction);
 
         //initialize again the widget's paper
         interaction.paper = widget.createPaper();
-        widget.createChoices(); 
+        widget.createChoices();
     };
 
-
     /**
      * Set up all elements to set the response mapping.
      * TODO this method needs to be split
      * @param {Oject} widget - the current widget
      */
-    function initResponseMapping(widget){
-        var scoreTexts      = {};
-        var interaction     = widget.element;
-        var $container      = widget.$original; 
-        var isResponsive    = $container.hasClass('responsive');
-        var $imageBox       = $('.main-image-box', $container);
-        var response        = interaction.getResponseDeclaration();
-        var mapEntries      = response.getMapEntries(); 
-        var corrects        = _.values(response.getCorrect());
+    function initResponseMapping(widget) {
+        const scoreTexts = {};
+        const interaction = widget.element;
+        const $container = widget.$original;
+        const isResponsive = $container.hasClass('responsive');
+        const $imageBox = $('.main-image-box', $container);
+        const response = interaction.getResponseDeclaration();
+        const mapEntries = response.getMapEntries();
+        let corrects = _.values(response.getCorrect());
 
         //get the shape of each choice
-        _.forEach(interaction.getChoices(), function(choice){    
-            var shape = interaction.paper.getById(choice.serial);
-            var $popup = grahicScorePopup(interaction.paper, shape, $imageBox, isResponsive);
-            var score = mapEntries[choice.id()] || response.mappingAttributes.defaultValue || '0'; 
+        _.forEach(interaction.getChoices(), function (choice) {
+            const shape = interaction.paper.getById(choice.serial);
+            const $popup = grahicScorePopup(interaction.paper, shape, $imageBox, isResponsive);
+            let scoreDefault = false;
+            if (mapEntries[choice.id()]) {
+                scoreDefault = false;
```

Comment:
I cannot see where this variable could be `true`, and it is always `false`.
Does it make sense to this conditional?

Reply:
Fixed @marpesia